src/Storages/ExternalStream/Kafka/Kafka.h

```diff
@@ -36,37 +60,53 @@ class Kafka final : public StorageExternalStreamImpl
     SinkToStoragePtr write(const ASTPtr & query, const StorageMetadataPtr & metadata_snapshot, ContextPtr context) override;
 
     bool produceOneMessagePerRow() const { return settings->one_message_per_row; }
+    Int32 topicRefreshIntervalMs() const { return topic_refresh_interval_ms; }
     const String & brokers() const { return settings->brokers.value; }
     const String & dataFormat() const override { return data_format; }
     const String & topic() const { return settings->topic.value; }
-    const klog::KConfParams & properties() const { return kafka_properties; }
-    const klog::KafkaWALAuth & auth() const noexcept { return *auth_info; }
     const ASTPtr & shardingExprAst() const { assert(!engine_args.empty()); return engine_args[0]; }
     bool hasCustomShardingExpr() const;
-    klog::KafkaWALSimpleConsumerPtr getConsumer(int32_t fetch_wait_max_ms = 200) const;
+
+    RdKafka::Producer & getProducer();
+    RdKafka::Topic & getProducerTopic();
```

Comment:
getTopic()

Reply:
This topic is a producer topic, it should be used to produce messages, because it's created with the `producer`. Calling it `getTopic()` will be incorrect.